src/metabase/query_processor/middleware/add_dimension_projections.clj

```diff
@@ -64,18 +75,23 @@
   get hidden when displayed anyway?)"
   [fields :- [mbql.s/Field]]
   (when-let [field-id->remapping-dimension (fields->field-id->remapping-dimension fields)]
-    ;; Reconstruct how we uniquify names in `metabase.query-processor.middleware.annotate`
-    (let [unique-name (comp (mbql.u/unique-name-generator) :name Field)]
+    ;; Reconstruct how we uniquify names in [[metabase.query-processor.middleware.annotate]]
+    (let [field       (fn [field-id]
+                        (qp.store/fetch-and-store-fields! #{field-id})
+                        (qp.store/field field-id))
+          unique-name (comp (mbql.u/unique-name-generator) :name field)]
       (vec
        (mbql.u/match fields
          ;; don't match Fields that have been joined from another Table
-         [:field (id :guard (every-pred integer? field-id->remapping-dimension)) (_ :guard (complement (some-fn :join-alias :source-field)))]
+         [:field (id :guard (every-pred integer? field-id->remapping-dimension)) _]
```

Comment:
You dropped the `(_ :guard (complement (some-fn :join-alias :source-field))` so its not clear to me that the comment about not matching Fields joined from another Table is still honored.

Reply:
you're right, that comment isn't true anymore. I'll remove it.